Not compatible with libcurl 7.35.0

Failed compile with libcurl 7.35.0.
It seems use obsoleted one.

My environment is 
Ubuntu 14.04 64bit
g++ 4.8.2

```
geniys@geni-u14-vm:~/Works/OSS/curlcpp/build$ cmake ..
-- The C compiler identification is GNU 4.8.2
-- The CXX compiler identification is GNU 4.8.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Found CURL: /usr/lib/x86_64-linux-gnu/libcurl.so (found suitable version "7.35.0", minimum required is "7.28.0") 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/geniys/Works/OSS/curlcpp/build
geniys@geni-u14-vm:~/Works/OSS/curlcpp/build$ make
Scanning dependencies of target curlcpp
[ 12%] Building CXX object src/CMakeFiles/curlcpp.dir/curl_easy.cpp.o
In file included from /home/geniys/Works/OSS/curlcpp/src/curl_easy.cpp:6:0:
/home/geniys/Works/OSS/curlcpp/include/curl_easy.h:276:31: error: ‘CURLOPT_OBSOLETE72’ was not declared in this scope
         CURLCPP_DEFINE_OPTION(CURLOPT_OBSOLETE72, long); /* OBSOLETE, do not use! */
                               ^
/home/geniys/Works/OSS/curlcpp/include/curl_easy.h:47:33: note: in definition of macro ‘CURLCPP_DEFINE_OPTION’
     template <> struct option_t<opt> {\
                                 ^
/home/geniys/Works/OSS/curlcpp/include/curl_easy.h:47:36: error: template argument 1 is invalid
     template <> struct option_t<opt> {\
                                    ^
/home/geniys/Works/OSS/curlcpp/include/curl_easy.h:276:9: note: in expansion of macro ‘CURLCPP_DEFINE_OPTION’
         CURLCPP_DEFINE_OPTION(CURLOPT_OBSOLETE72, long); /* OBSOLETE, do not use! */
         ^
make[2]: *** [src/CMakeFiles/curlcpp.dir/curl_easy.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/curlcpp.dir/all] Error 2
make: *** [all] Error 2
geniys@geni-u14-vm:~/Works/OSS/curlcpp/build$ g++ --version
g++ (Ubuntu 4.8.2-19ubuntu1) 4.8.2
Copyright (C) 2013 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

geniys@geni-u14-vm:~/Works/OSS/curlcpp/build$
```

Thanks

--- a/include/curl_easy.h
+++ b/include/curl_easy.h
@@ -273,7 +273,12 @@ namespace curl  {
         /* Max amount of cached alive connections */
         CURLCPP_DEFINE_OPTION(CURLOPT_MAXCONNECTS, long);
 
+        /* Name changed after 7.37.0 */
+#if defined(LIBCURL_VERSION_NUM) && LIBCURL_VERSION_NUM > 0x072500
         CURLCPP_DEFINE_OPTION(CURLOPT_OBSOLETE72, long); /* OBSOLETE, do not use! */
+#else
+        CURLCPP_DEFINE_OPTION(CURLOPT_DEPRECATED72, long);
+#endif
 
         /* 73 = OBSOLETE */
 
@@ -705,8 +710,8 @@ namespace curl  {
         CURLCPP_DEFINE_OPTION(CURLOPT_INTERLEAVEDATA, void*);
 
         /* Let the application define a custom write method for RTP data */
-        CURLCPP_DEFINE_OPTION(CURLOPT_INTERLEAVEFUNCTION, size_t(*)(void *ptr, 
-            size_t size, 
+        CURLCPP_DEFINE_OPTION(CURLOPT_INTERLEAVEFUNCTION, size_t(*)(void *ptr,
+            size_t size,
             size_t nmemb,
             void *userdata));
 
@@ -716,7 +721,7 @@ namespace curl  {
         /* Directory matching callback called before downloading of an
         individual file (chunk) started */
         CURLCPP_DEFINE_OPTION(CURLOPT_CHUNK_BGN_FUNCTION, long(*)(
-            const void *transfer_info, 
+            const void *transfer_info,
             void *ptr,
             int remains));
 
@@ -870,7 +875,7 @@ namespace curl  {
     }  // of namespace detail
 
     /**
-     * Easy interface is used to make requests and transfers. 
+     * Easy interface is used to make requests and transfers.
      * You don't have to worry about freeing data or things like
      * that. The class will do it for you.
      */
@@ -946,7 +951,7 @@ namespace curl  {
          */
         unique_ptr<vector<string>> get_info(const CURLINFO) const;
         /**
-         * Using this function, you can explicitly pause a running connection, 
+         * Using this function, you can explicitly pause a running connection,
          * and you can resume a previously paused connection.
          */
         void pause(const int);
@@ -978,7 +983,7 @@ namespace curl  {
     private:
         CURL *curl;
     };
-    
+
     // Implementation of overloaded add method.
     template<typename Iterator> void curl_easy::add(Iterator begin, const Iterator end) {
         for (; begin != end; ++begin) {
@@ -992,7 +997,7 @@ namespace curl  {
             throw curl_easy_exception(code, __FUNCTION__);
         }
     }
-    
+
     // Implementation of add method.
     template<typename T> void curl_easy::add(const curl_pair<CURLoption,T> pair) {
         const CURLcode code = curl_easy_setopt(this->curl,pair.first(),pair.second());

